core/clustersmngr/altclient.go

```diff
@@ -0,0 +1,180 @@
+package clustersmngr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/weaveworks/weave-gitops/core/nsaccess"
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NewAltClient() (Client, error) {
+
+	return &altClient{}, nil
+}
+
+type altClient struct {
+	Client
+}
+
+type GitopsClusterCRSpec struct {
+	SecretRef map[string]string `json:"secretRef"`
+}
+
+type GitopsClusterCR struct {
+	Spec GitopsClusterCRSpec `json:"spec"`
+}
+
+func (ac *altClient) ClusteredList(ctx context.Context, clist ClusteredObjectList, namespaced bool, opts ...client.ListOption) error {
+	// build a client from values extracted from ctx or read from cache
+	fmt.Println("doing other client")
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return fmt.Errorf("building in-cluster config: %w", err)
+	}
+
+	user := auth.Principal(ctx)
+
+	cfg.BearerToken = user.Token()
```

Comment:
We might also have to `cfg.BearerTokenFile = ""` (which I guess is the ServiceAccount), as it seems to take precedence (https://github.com/kubernetes/client-go/blob/master/transport/round_trippers.go#L289)